amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/jsonschema/AstIndexBuilder.scala

```diff
@@ -28,23 +37,25 @@ object AstIndexBuilder {
 
   private def getResolutionScope(version: SchemaVersion, baseUri: URI): Option[ResolutionScope] = version match {
     case JSONSchemaDraft3SchemaVersion | JSONSchemaDraft4SchemaVersion | JSONSchemaDraft6SchemaVersion =>
-      Some(Draft4ResolutionScope(baseUri))
-    case JSONSchemaDraft7SchemaVersion      => Some(Draft7ResolutionScope(baseUri))
+      Some(Draft4IdResolutionScope(baseUri))
+    case JSONSchemaDraft7SchemaVersion      => Some(Draft7IdResolutionScope(baseUri))
     case JSONSchemaDraft201909SchemaVersion => Some(Draft2019ResolutionScope(baseUri))
     case _                                  => None
   }
 }
 
 case class AstIndexBuilder private (private val refCounter: AliasCounter)(implicit ctx: WebApiContext) {
 
-  def build(node: YNode, scopes: Seq[ResolutionScope]): AstIndex = {
+  def build(node: YNode, scopes: Seq[ResolutionScope], resolvers: Seq[ReferenceResolver]): AstIndex = {
     val acc = mutable.Map.empty[String, YMapEntryLike]
     scopes.foreach(_.resolve("", YMapEntryLike(node), acc))
     index(YMapEntryLike(node), scopes, acc)
-    AstIndex(acc)
+    AstIndex(acc, resolvers)
```

Comment:
to avoid passing resolvers as parameter to function, we could create an AstIndex with no resolvers and then `copy()` the AstIndex as it is a case class with the resolvers we want to add.

Reply:
I believe passing the parameter of resolvers explicitly might be helpful to understand which elements are needed for the creation of the AstIndex.